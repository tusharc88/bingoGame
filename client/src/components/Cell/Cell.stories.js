import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered';
import { withKnobs, number } from '@storybook/addon-knobs';
import Cell from './index';

storiesOf('Cell', module)
  .addDecorator(withKnobs)
  .addDecorator(centered)
  .addWithInfo('default', <Cell>1</Cell>, () => (
    <Cell currentNumber={number('currentNumber', 1)}>
      {number('children', 1)}
    </Cell>
  ));
