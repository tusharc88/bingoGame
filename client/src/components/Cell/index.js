import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CellWrapper = styled.div`
  height: 40px;
  width: 40px;
  background-color: ${props => (props.isClicked ? '#3ecedf' : '#d7d3d2')};
  color: ${props => (props.isClicked ? '#ffffff' : '#393838')};
  font-size: 25px;
  text-align: center;
  vertical-align: middle;
  line-height: 40px;
  border: 1px solid #ffffff;
  transition: ${props => !props.disabled && 'all 0.3s ease-in-out'};
  transition-property: ${props =>
    !props.disabled && 'box-shadow, color, background-color'};
  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
  ${props =>
    !props.disabled &&
    `&:hover {
    box-shadow: ${`inset 0 0 0 3px ${props.isClicked ? '#d7d3d2' : '#3ecedf'}`};
    transition: 'all 0.3s ease-in-out';
    transition-property: 'box-shadow, color, background-color';
  }`};
`;

export class Cell extends Component {
  static propTypes = {
    children: PropTypes.number.isRequired,
    currentNumber: PropTypes.number.isRequired,
  };

  state = {
    isClicked: false,
  };

  handleClick = () => {
    if (!this.handleDisabled()) {
      this.setState(state => ({
        ...state,
        isClicked: !this.state.isClicked,
      }));
    }
  };

  handleDisabled = () => this.props.currentNumber !== this.props.children;

  render() {
    const {
      props: { children },
      state: { isClicked },
    } = this;

    return (
      <CellWrapper
        onClick={this.handleClick}
        isClicked={isClicked}
        disabled={this.handleDisabled()}
      >
        {children}
      </CellWrapper>
    );
  }
}

export default Cell;
