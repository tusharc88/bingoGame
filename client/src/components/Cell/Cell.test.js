import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import 'jest-styled-components';
import Cell from './index';

const wrap = (props = {}) =>
  shallow(
    <Cell currentNumber={1} {...props}>
      {1}
    </Cell>
  );

let wrapper = null;
let tree = null;

beforeAll(() => {
  wrapper = wrap();
  tree = toJSON(wrapper);
});

test('Renders with default props', () => {
  expect(tree).toMatchSnapshot();
});

test('handleDisabled should return false when currentNumber is 1 and Cell number is 1', () => {
  expect(wrapper.instance().handleDisabled()).toEqual(false);
});

test('handleDisabled should return true when currentNumber is 2 and Cell number is 1', () => {
  const newWrap = (props = {}) =>
    shallow(
      <Cell currentNumber={2} {...props}>
        {1}
      </Cell>
    );

  const newWrapper = newWrap();
  expect(newWrapper.instance().handleDisabled()).toEqual(true);
});

test('handleClick should toggle background color on clicking non disabled cell', () => {
  wrapper.simulate('click');
  expect(wrapper).toHaveStyleRule('background-color', '#3ecedf');
  wrapper.simulate('click');
  expect(wrapper).toHaveStyleRule('background-color', '#d7d3d2');
});

test('handleClick should not toggle background color on clicking disabled cell', () => {
  const newWrap = (props = {}) =>
    shallow(
      <Cell currentNumber={2} {...props}>
        {1}
      </Cell>
    );

  const newWrapper = newWrap();

  newWrapper.simulate('click');
  expect(newWrapper).toHaveStyleRule('background-color', '#d7d3d2');
});
