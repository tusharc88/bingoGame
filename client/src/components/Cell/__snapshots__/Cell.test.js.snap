// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with default props 1`] = `
<styled.div
  disabled={false}
  isClicked={false}
  onClick={[Function]}
>
  1
</styled.div>
`;
